prow/kube/client.go

```diff
@@ -181,8 +183,11 @@ func (c *Client) requestRetry(r *request) ([]byte, error) {
 	return rb, nil
 }
 
-func (c *Client) doRequest(method, urlPath string, query map[string]string, body interface{}) (*http.Response, error) {
+func (c *Client) doRequest(method, deckPath, urlPath string, query map[string]string, body interface{}) (*http.Response, error) {
 	url := c.baseURL + urlPath
+	if c.deckURL != "" {
```

Comment:
I think this should be `if c.deckURL != "" && deckPath != "" {`
Otherwise if plank specifies a `deckURL` it won't be able to update or create prowjobs.

Reply:
Done.